packages/common/src/ui/forms/Modal/ModalAutocomplete.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { PropsWithChildren } from 'react';
+import {
+  Autocomplete,
+  AutocompleteChangeDetails,
+  AutocompleteRenderInputParams,
+  AutocompleteRenderOptionState,
+  createFilterOptions,
+  Grid,
+  TextField,
+  Theme,
+} from '@mui/material';
+
+export interface ModalAutocompleteProps<T> {
+  filterOptions: any;
+  loading?: boolean;
+  limitTags?: number;
+  loadingText?: React.ReactNode;
+  noOptionsText?: React.ReactNode;
+  onChange?: (
+    event: React.SyntheticEvent,
+    value: T | null,
+    reason: string,
+    details?: AutocompleteChangeDetails<T>
+  ) => void;
+  options: readonly T[];
+  renderInput?: (params: AutocompleteRenderInputParams) => React.ReactNode;
+  width?: number;
+  renderOption?: (
+    props: React.HTMLAttributes<HTMLLIElement>,
+    option: T,
+    state: AutocompleteRenderOptionState
+  ) => React.ReactNode;
+}
+
+const inputStyle = {
+  backgroundColor: (theme: Theme) => theme.palette.background.menu,
+  borderRadius: '8px',
+  padding: '4px 8px',
+  '& .MuiInput-underline:before': { borderBottomWidth: 0 },
+  '& .MuiInput-input': { color: (theme: Theme) => theme.palette.darkGrey },
+};
+
+export { createFilterOptions };
+
+export function ModalAutocomplete<T>({
+  filterOptions,
+  loading,
+  loadingText,
+  noOptionsText,
+  onChange,
+  options,
+  renderInput,
+  renderOption,
+  width,
+}: PropsWithChildren<ModalAutocompleteProps<T>>): JSX.Element {
```

Comment:
Could be nice to expose the filter options as props `filterStringifiererer` etc and having the component construct the options rather than consumers having to import the filter factory also?

Reply:
yeah, much better. will do